packages/medusa/src/loaders/models.ts

```diff
@@ -5,30 +5,76 @@ import { ClassConstructor, MedusaContainer } from "../types/global"
 import { EntitySchema } from "typeorm"
 import { asClass, asValue } from "awilix"
 
+function formatModelName(modelName) {
+  // Remove "Model" from the end of the string
+  let formattedName = modelName.replace('Model', '');
+
+  // Capitalize the first letter of the string
+  formattedName = formattedName.charAt(0).toUpperCase() + formattedName.slice(1);
+
+  return formattedName;
+}
+
 /**
  * Registers all models in the model directory
  */
 export default (
-  { container, isTest }: { container: MedusaContainer; isTest?: boolean },
+  { container, isTest, rootDirectory }: { container: MedusaContainer; isTest?: boolean, rootDirectory?: string },
   config = { register: true }
 ) => {
   const corePath = isTest ? "../models/*.ts" : "../models/*.js"
+  const modelExtensionsGlob = "dist/models/*.js"
   const coreFull = path.join(__dirname, corePath)
-
+  const modelExtensionsFullGlob = rootDirectory ? path.join(rootDirectory, modelExtensionsGlob) : null
   const models: (ClassConstructor<unknown> | EntitySchema)[] = []
 
   const core = glob.sync(coreFull, {
     cwd: __dirname,
     ignore: ["index.js", "index.ts"],
   })
+
+  const modelExtensions = modelExtensionsFullGlob ? glob.sync(modelExtensionsFullGlob, {
+    ignore: ["index.js"],
```

Comment:
todo: add index.ts as well just in case

Reply:
the glob looks only for js files, it will be excluded on the glob level.